cirq/devices/grid_qubit.py

```diff
@@ -12,43 +12,27 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
+import functools
+from typing import Iterable, List, Optional, Tuple, Set, TypeVar, TYPE_CHECKING
 
-from typing import Iterable, List, Optional, Set, Tuple, TYPE_CHECKING
+import abc
 
 from cirq import ops, protocols
 
 if TYPE_CHECKING:
     import cirq
 
+TSelf = TypeVar('TSelf', bound='_BaseGridQid')
 
-class GridQubit(ops.Qid):
-    """A qubit on a 2d square lattice.
-
-    GridQubits use row-major ordering:
-
-        GridQubit(0, 0) < GridQubit(0, 1) < GridQubit(1, 0) < GridQubit(1, 1)
-
-    New GridQubits can be constructed by adding or subtracting tuples
-
-        >>> cirq.GridQubit(2, 3) + (3, 1)
-        cirq.GridQubit(5, 4)
 
-        >>> cirq.GridQubit(2, 3) - (1, 2)
-        cirq.GridQubit(1, 1)
-    """
+@functools.total_ordering
```

Comment:
Why `total_ordering`?

Reply:
jbc that's what _BaseLineQid uses; I was confused bc it seemed like Qid implements all the rich comparison functions, but figured there was a reason for it to be there: possibly removable from both _BaseLineQid and _BaseGridQid?

Reply:
It might be a performance optimization somehow. But normally it requires you to define e.g. `__lt__`